README.md

```diff
@@ -1392,6 +1394,266 @@ OPTIONS:
             The file name
 ```
 
+## bdev
+
+The `bdev` commands are for interacting with block devices in Flash. These can either be defined using binary info, or you can specify a partition to use as a block device. They work with a LittleFS filesystem, or a FatFS filesystem.
+
+```text
+$ picotool help bdev
+BDEV:
+    Commands related to embedded block devices
+
+SYNOPSIS:
+    picotool bdev ls <dirname> [-r] [-p <partition>] [--filesystem <fs>] [--format] [device-selection]
+    picotool bdev mkdir <dirname> [-p <partition>] [--filesystem <fs>] [--format] [device-selection]
+    picotool bdev cp <src> <dest> [-p <partition>] [--filesystem <fs>] [--format] [device-selection]
+    picotool bdev rm <filename> [-p <partition>] [--filesystem <fs>] [--format] [device-selection]
+    picotool bdev cat <filename> [-p <partition>] [--filesystem <fs>] [--format] [device-selection]
+
+SUB COMMANDS:
+    ls      List contents of the block device
+    mkdir   Create directory on the block device
+    cp      Copy file to/from the block device - use :filename to indicate files on the device (eg `cp main.py :main.py` to upload to the
+            device)
+    rm      Delete a file or an empty directory on the block device
+    cat     Print contents of file on the block device
+```
+
+### ls
+
+List contents of the block device
+
+```text
+$ picotool help bdev ls
+BDEV LS:
+    List contents of the block device
+
+SYNOPSIS:
+    picotool bdev ls <dirname> [-r] [-p <partition>] [--filesystem <fs>] [--format] [device-selection]
+
+OPTIONS:
+        <dirname>
+            The directory name to list (optional)
+        -r, --recursive
+            List files in directories recursively
+    Block device options
+        -p, --partition
+            Partition to use as block device
+        <partition>
+            partition number
+        --filesystem
```

Comment:
Is the `--filesystem` flag only needed when formatting a block-device, or are there likely to be circumstances in which the filesystem auto-deteciton doesn't work?

Reply:
With the current supported filesystems it should only be needed when formatting, but future filesystems may be harder to auto-detect so may require this flag